### PR TITLE
Warn on CLI about experimental backends

### PR DESCRIPTION
**Pull request by jschneider-bensch**
This will emit a warning message when using the EasyCrypt or ProVerif backends, addressing #358.
Hope I found the appropriate place for the warning code!

### CODE REVIEW

File: cli/driver/src/exporter.rs [hunk 1]
```diff
@@ -390,6 +390,16 @@ impl Callbacks for ExtractionCallbacks {
...
+                    match backend.backend {
+                        hax_cli_options::Backend::Easycrypt
+                        | hax_cli_options::Backend::ProVerif => {
+                            eprintln!(
+                                "⚠️ Warning: Experimental backend \"{}\" is work in progress.",
+                                backend.backend
+                            )
+                        }
+                        _ => (),
+                    };
```
Comment (W95Psp) [inline cli/driver/src/exporter.rs:402]: Small (purely syntactical) nit: you could write that with the [`matches` macro](https://doc.rust-lang.org/std/macro.matches.html)
```rust
use hax_cli_options::Backend;
if matches!(backend.backend, Backend::Easycrypt | Backend::ProVerif) {
  ...
}

**Review by W95Psp — APPROVED** [review]
LGTM, just a small nit (but feel free to ignore it, it's quite subjective)

File: cli/driver/src/exporter.rs (hunk 1 above)
Comment (jschneider-bensch) [reply cli/driver/src/exporter.rs:402]: Thanks for pointing that out! That's much more concise. :+1:

Comment (W95Psp) [reply cli/driver/src/exporter.rs]: :smiley:

**Pull request merged**